src/source/PeerConnection/PeerConnection.c

```diff
@@ -1826,47 +1848,88 @@ STATUS deinitKvsWebRtc(VOID)
     return retStatus;
 }
 
-STATUS twccManagerOnPacketSent(PKvsPeerConnection pc, PRtpPacket pRtpPacket)
+// Not thread safe. Ensure this function is invoked in a guarded section
+static STATUS twccRollingWindowDeletion(PKvsPeerConnection pKvsPeerConnection, PRtpPacket pRtpPacket, UINT16 seqNum)
+{
```

Comment:
Thinking we may also want to add a regression test to ensure the that deletion behavior doesn't have any no dangling elements in the hash table and `firstSeqNumInRollingWindow` is updated correctly?

Reply:
Will add additional unit tests in a future PR per offline discussion.